src/omics/tests/unit/test_get_omics_log.py

```diff
@@ -0,0 +1,90 @@
+from unittest.mock import Mock, patch
+
+import pytest
+from ugbio_omics.get_omics_log import OMICS_LOG_GROUP, fetch_save_log, get_log_for_task
+
+
+@pytest.fixture
+def mock_boto3_client():
+    with patch("boto3.client") as mock_client:
+        # Create separate mock clients for "logs" and "omics"
+        mock_logs_client = Mock()
+        mock_omics_client = Mock()
+
+        # Configure the side_effect to return the appropriate mock client based on the service name
+        def client_side_effect(service_name, *args, **kwargs):
+            if service_name == "logs":
+                return mock_logs_client
+            elif service_name == "omics":
+                return mock_omics_client
+            else:
+                raise ValueError(f"Unexpected service name: {service_name}")
+
+        mock_boto3_client.side_effect = client_side_effect
+
+        # Configure the mock clients
+        mock_logs_client.get_log_events.return_value = {
+            "events": [{"message": "log message 1"}, {"message": "log message 2"}]
+        }
+        yield mock_client
+
+
+@pytest.fixture
+def mock_get_run_info():
+    with patch("ugbio_omics.get_omics_log.get_run_info") as mock_get_run_info:
+        mock_get_run_info.return_value = {
+            "tasks": [
+                {"taskId": "task1", "name": "Task_1", "status": "SUCCEEDED"},
+                {"taskId": "task2", "name": "Task_2", "status": "FAILED"},
+            ],
+            "status": "SUCCEEDED",
+        }
+        yield mock_get_run_info
+
+
+def test_get_log_for_task_all_tasks(mock_boto3_client, mock_get_run_info, tmpdir):
+    get_log_for_task("run1", output_path=tmpdir, output_prefix="test_")
+
+    mock_get_run_info.assert_called_once()
+    mock_boto3_client.assert_called_with("logs")
+    assert len(tmpdir.listdir()) == 2  # check that 2 log files created for all tasks
+
+
+def test_get_log_for_task_failed_tasks(mock_boto3_client, mock_get_run_info, tmpdir):
+    get_log_for_task("run1", output_path=tmpdir, output_prefix="test_", exclude_failed=True)
+
+    mock_get_run_info.assert_called_once()
+    mock_boto3_client.assert_called_with("logs")
+    assert len(tmpdir.listdir()) == 1  # check that one log file created for the succesful task
+
```

Comment:
can we add test for the use case that there're no failed task and we get the engine log?